charts/skypilot/templates/ingress.yaml

```diff
@@ -18,10 +18,12 @@ metadata:
     # Bearer token requests bypass OAuth2 proxy and go directly to the application
     nginx.ingress.kubernetes.io/auth-signin: {{ if index .Values.ingress "oauth2-proxy" "use-https" | default false }}https{{ else }}http{{ end }}://$host/oauth2/start?rd=$escaped_request_uri
     nginx.ingress.kubernetes.io/auth-url: {{ if index .Values.ingress "oauth2-proxy" "use-https" | default false }}https{{ else }}http{{ end }}://$host/oauth2/auth
-    nginx.ingress.kubernetes.io/auth-response-headers: X-Auth-Request-Email
+    nginx.ingress.kubernetes.io/auth-response-headers: X-Auth-Request-Email, X-Skypilot-Auth
     # Skip OAuth2 authentication for requests with Authorization header (Bearer tokens)
     nginx.ingress.kubernetes.io/auth-snippet: |
-      if ($http_authorization ~ "^Bearer .+") {
+      if ($http_authorization ~ "^Bearer sky_.+") {
+        # Set x_skypilot_auth_mode variable to make sure the request is validated on the API server side.
+        set $x_skypilot_auth_mode "token";
         return 200;
       }
```

Comment:
just curious, is it possible to ban connections from users with the `X_SKYPILOT_AUTH_MODE` set to `"token"` already? In that case, our ingress becomes the only one who can turn on the token mode -- might be more secure : )

Reply:
Will do it in a followup PR!